src/apps/config/alarm_codec.lua

```diff
@@ -184,7 +196,7 @@ function selftest ()
       assert(lib.equal({name, args}, decoded))
    end
    local function test_raise_alarm ()
-      local key = {resource='res1'}
+      local key = {resource='res1', alarm_type_id='type1', alarm_type_qualifier=''}
```

Comment:
Possibly revert this change so that we are testing the right thing?  And below.

Reply:
I left `resource`, `alarm_type_id` and `alarm_type_qualifier` encoded as strings, because according to the YANG schema they are mandatory fields (I considered though whether to give `alarm_type_qualifier` a default empty string value in case of `nil`, because I believe many times this value won't be used)

The `maybe_string` functionality is tested via `local args`. In this case, all arguments in `args` might be nil. The test only uses `perceived_severity` and it's working fine, so that means `maybe_string` is tested and working OK.